### PR TITLE
Update: separate options for block and line comments in `spaced-comment` rule

### LINKED ISSUE #2897: New spaced-comment rule to only enforce space after //
Is it no longer possible to only enforce space after // but not after /* ?

### PR DESCRIPTION
**Pull request by BYK**
Fixes #2897

### CODE REVIEW

File: docs/rules/spaced-comment.md [hunk 1]
```diff
@@ -41,6 +41,12 @@ The `"markers"` array will apply regardless of the value of the first argument,
 The difference between a marker and an exception is that a marker only appears at the beginning of the comment whereas
 exceptions can occur anywhere in the comment string.
 
+You can also define separate exceptions and markers for block and line comments:
+
+```js
+"spaced-comment": [2, "always", { "line": { "markers": ["/"] }, "block": { "markers": ["*"] } }]
```
Comment (nzakas) [inline docs/rules/spaced-comment.md]: You can also do `exceptions` for each, correct? We should list out all possible options in this example (and maybe use some formatting to spread out across a few lines?).

File: lib/rules/spaced-comment.js [hunk 2]
```diff
@@ -158,7 +152,9 @@ module.exports = function(context) {
      * @returns {void}
      */
     function checkCommentForSpace(node) {
-        var commentIdentifier = node.type === "Block" ? "/*" : "//";
+        var type = node.type.toLowerCase();
```
Comment (nzakas) [inline lib/rules/spaced-comment.js]: Can you combine these `var` statements please? :)

File: tests/lib/rules/spaced-comment.js [hunk 3]
```diff
@@ -349,6 +375,52 @@ ruleTester.run("spaced-comment", rule, {
...
+            }]
+        },
+        {
+            code: "///--------\r\n/// test\r\n///--------\r\n/*/ blah */",
+            options: ["always", { markers: ["/"], exceptions: ["-"], block: { markers: [] } }],
+            errors: [{
+                message: "Expected exception block, space or tab after /* in comment.",
+                type: "Block"
+            }]
         }
```
Comment (nzakas) [inline tests/lib/rules/spaced-comment.js]: Can you add some tests that contain both line comments and block comments so you can be sure that block comment settings aren't affecting line comments and vice-versa?

Comment (BYK) [inline tests/lib/rules/spaced-comment.js]: This last one is exactly what you ask for actually :)

**Comment by BYK** [comment]
@nzakas - Updated. Moar feedback welcome (especially around tests)

**Comment by nzakas** [comment]
LGTM

**Pull request merged**